sharding/collator/collator.go

```diff
@@ -56,21 +56,22 @@ func subscribeBlockHeaders(c client.Client) error {
 // conditions are met
 func checkSMCForCollator(c client.Client, head *types.Header) error {
 	log.Info("Checking if we are an eligible collation collator for a shard...")
-	period := big.NewInt(0).Div(head.Number, big.NewInt(sharding.PeriodLength))
 	for s := int64(0); s < sharding.ShardCount; s++ {
 		// Checks if we are an eligible collator according to the SMC
-		addr, err := c.SMCCaller().GetEligibleCollator(&bind.CallOpts{}, big.NewInt(s), period)
+		for i := int64(0); i < sharding.NotaryCommitSize; s++ {
+			addr, err := c.SMCCaller().GetNotaryInCommittee(&bind.CallOpts{}, big.NewInt(s), big.NewInt(i))
```

Comment:
What is going on in this method? 
According to the method name and comment, this was supposed to check if we are an eligible collator, but it is checking to see if we are a notary?

Reply:
It's checking if we are a notary that's in the committee for the given shard and index

Reply:
My concerns about this are addressed in #96